src/lib.rs

```diff
@@ -171,17 +195,31 @@ impl DateTime {
         Self::from_utc(dt.assume_utc())
     }
 
-    /// Convert to UNIX timestamp
-    pub const fn into_unix_timestamp(self) -> i64 {
-        let Self { unix_timestamp } = self;
-        unix_timestamp
+    /// Convert to UNIX timestamp in seconds.
+    pub const fn into_timestamp_secs(self) -> i64 {
+        self.timestamp_micros / 1_000_000
+    }
+
+    /// Convert to UNIX timestamp in milliseconds.
+    pub const fn into_timestamp_millis(self) -> i64 {
+        self.timestamp_micros / 1_000
+    }
+
+    /// Convert to UNIX timestamp in microseconds.
+    pub const fn into_timestamp_micros(self) -> i64 {
+        self.timestamp_micros
+    }
+
+    /// Convert to UNIX timestamp in microseconds.
+    pub const fn into_timestamp_nanos(self) -> i64 {
```

Comment:
i think we should ditch the nanos one. It could suggest we have nanos precision.